lib/RestructureCFG/InlineDispatcherSwitch.cpp

```diff
@@ -509,6 +509,65 @@ static void processNestedWeavedSwitches(SwitchNode *Switch) {
   }
 }
 
+static RecursiveCoroutine<bool> containsNonLocalCF(ASTNode *Node) {
+  switch (Node->getKind()) {
+  case ASTNode::NK_List: {
+    SequenceNode *Seq = llvm::cast<SequenceNode>(Node);
+
+    bool HasNonLocalCF = false;
+    for (ASTNode *N : Seq->nodes()) {
+      HasNonLocalCF = HasNonLocalCF or rc_recur containsNonLocalCF(N);
+    }
+
+    rc_return HasNonLocalCF;
+  } break;
+  case ASTNode::NK_Scs: {
+
+    // We do not want to move around a `ScsNode` for any reason
```

Comment:
We don't because we already know it creates problems? Or it's just an advanced scenario we haven't fully considered yet?

Reply:
There was actually a bug here, the inlining was always enabled, I now disabled it [here](https://github.com/revng/revng-c/blob/f15b7b1a9506a37ed6d348f7dd6d880b1d93df16/lib/RestructureCFG/InlineDispatcherSwitch.cpp#L532), and added a TODO mentioning that we may want to extend the inlining criterion to cover also this scenario.

The main point is that we could consider this advanced scenario at the cost of moving also the eventual exit dispatcher associated to a cycle when moving the cycle itself. If we do that, I think that we may also enable the inlining for `ScsNode`s.

Reply:
> The main point is that we could consider this advanced scenario at the cost of moving also the eventual exit dispatcher associated to a cycle when moving the cycle itself. If we do that, I think that we may also enable the inlining for ScsNodes.

Ok then please expand the comments explaining this. Otherwise, like it is now, it seems that you never want to do it, not that it's a trade-off decision.

Reply:
I motivated this choice, and highlighted that we may want to reverse it, [here](https://github.com/revng/revng-c/blob/6af6c8ffa81d1d570010e6e64e260a78eedd4ea9/lib/RestructureCFG/InlineDispatcherSwitch.cpp#L540)